src/datanode/src/sql/copy_table.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::pin::Pin;
+
+use common_query::physical_plan::SessionContext;
+use common_query::Output;
+use common_recordbatch::adapter::DfRecordBatchStreamAdapter;
+use datafusion::parquet::arrow::ArrowWriter;
+use datafusion::parquet::basic::{Compression, Encoding};
+use datafusion::parquet::file::properties::WriterProperties;
+use datafusion::physical_plan::RecordBatchStream;
+use futures::TryStreamExt;
+use object_store::services::fs::Builder;
+use object_store::ObjectStore;
+use snafu::ResultExt;
+use table::engine::TableReference;
+use table::requests::CopyTableRequest;
+
+use crate::error::{self, Result};
+use crate::sql::SqlHandler;
+
+impl SqlHandler {
+    pub(crate) async fn copy_table(&self, req: CopyTableRequest) -> Result<Output> {
+        let table_ref = TableReference {
+            catalog: &req.catalog_name.to_string(),
+            schema: &req.schema_name.to_string(),
+            table: &req.table_name.to_string(),
+        };
+        let table = self.get_table(&table_ref)?;
+
+        let stream = table
+            .scan(None, &[], None)
+            .await
+            .with_context(|_| error::CopyTableSnafu {
+                table_name: table_ref.to_string(),
+            })?;
+
+        let stream = stream
+            .execute(0, SessionContext::default().task_ctx())
+            .context(error::TableScanExecSnafu)?;
+        let stream = Box::pin(DfRecordBatchStreamAdapter::new(stream));
+
+        let accessor = Builder::default().build().unwrap();
+        let object_store = ObjectStore::new(accessor);
+
+        let mut parquet_writer = ParquetWriter::new(req.file_name, stream, object_store);
+        let rows = parquet_writer.flush().await?;
```

Comment:
Seems `COPY` is implemented synchronously. When copying large table, it's likely to block client for long time. Maybe we should make "copy" runs in background? Like pg: https://www.postgresql.org/docs/current/sql-copy.html

Reply:
Asynchronous (background) task export seems to be a complex task, like progress view of background tasks (need to store some state), control concurrency, etc. So I prefer to implement these in multiple tasks. I can leave a TODO for now, and will seriously consider how to implement it well in the future.

The current goal and requirement is that it can reliably perform data export (for backup) and import data into DB in a synchronized way.